src/Language/PureScript/Docs/Convert/ReExports.hs

```diff
@@ -123,72 +122,22 @@ collectDeclarations imports exports = do
   collect
     :: (Eq a, Show a)
     => (P.ModuleName -> a -> m (P.ModuleName, [b]))
-    -> [P.ImportRecord a]
     -> Map a P.ExportSource
     -> m (Map P.ModuleName [b])
-  collect lookup' imps exps = do
-    imps' <- traverse (findImport imps) $ Map.toList $ fmap P.exportSourceDefinedIn exps
-    Map.fromListWith (<>) <$> traverse (uncurry lookup') imps'
+  collect lookup' exps =
+    exps
+    |> Map.mapMaybe P.exportSourceImportedFrom
+    |> Map.toList
+    |> traverse (uncurry (flip lookup'))
+    |> fmap (Map.fromListWith (<>))
```

Comment:
I kinda prefer the way this was written before, mainly because this would be the only place in the whole compiler that this style is used that I'm aware of. (The other part is because I'm one of those weird people that finds this ordering harder to parse than it being "backwards" as "normal" :smile:).

Reply:
Yes, totally fair point, I'll use something closer to the older style :)